category_theory/yoneda.lean

```diff
@@ -0,0 +1,107 @@
+-- Copyright (c) 2017 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison
+
+/- The Yoneda embedding, as a functor `yoneda : C ⥤ ((Cᵒᵖ) ⥤ (Type v₁))`,
+   along with instances that it is `full` and `faithful`.
+   
+   Also the Yoneda lemma, `yoneda_lemma : (yoneda_pairing C) ≅ (yoneda_evaluation C)`. -/
+
+import category_theory.natural_transformation
+import category_theory.opposites
+import category_theory.types
+import category_theory.embedding
+import category_theory.natural_isomorphism
+
+namespace category_theory
+
+universes u₁ v₁ u₂
+
+variables (C : Type u₁) [𝒞 : category.{u₁ v₁} C]
+include 𝒞
+
+def yoneda : C ⥤ ((Cᵒᵖ) ⥤ (Type v₁)) := 
+{ obj := λ X,
+  { obj := λ Y : C, Y ⟶ X,
+    map' := λ Y Y' f g, f ≫ g,
+    map_comp' := begin intros X_1 Y Z f g, ext1, dsimp at *, erw [category.assoc] end,
+    map_id' := begin intros X_1, ext1, dsimp at *, erw [category.id_comp] end },
+  map' := λ X X' f, { app := λ Y g, g ≫ f } }
+
+namespace yoneda
+@[simp] lemma obj_obj (X Y : C) : ((yoneda C) X) Y = (Y ⟶ X) := rfl
+@[simp] lemma obj_map (X : C) {Y Y' : C} (f : Y ⟶ Y') : ((yoneda C) X).map f = λ g, f ≫ g := rfl
+@[simp] lemma map_app {X X' : C} (f : X ⟶ X') (Y : C) : ((yoneda C).map f) Y = λ g, g ≫ f := rfl
+
+lemma aux_1 {X Y : Cᵒᵖ} (f : X ⟶ Y) : ((yoneda C) X).map f (𝟙 X) = ((yoneda C).map f) Y (𝟙 Y) := 
```

Comment:
still `aux_1` and `aux_2`.

what about `aux_1` => `map_id`
and `aux_2` => `app`?

Reply:
Okay, I've renamed these `yoneda.obj_map_id` and `yoneda.naturality`.